### PR TITLE
Improvement on TimoshenkoFPM and IGA

### PR DESCRIPTION
**Pull request by pchaoWT**
Several improvement on Timoshenko with fully-populated matrix(FPM) and IGA from last pull request:
1. Complete the doxygen-format comments for Timoshenko beam and several other codes. You could generate the doxygen-based document easily.
2. Develop a new Timoshenko beam in which the mass,stiffness and damping matrices are  evaluated via Guass quadrature, also the fully-populated matrix(FPM) is supported.
3. Enable visualization of FPM Timoshenko beams, but I haven't done any test on this commit.
4. Add a new class in ChBeamSectionCosserat to allow input FPM directly for IGA beam. This has been included in last pull request, but maybe in this pull request, I have done some improvement, such as the doxygen-format comments,etc. 
5. Declaration: the code format is automatically reformated by visual studio IDE since the .clang-format is found by IDE. I'm not sure whether the present format is reasonable. If not, I'm sorry for this unconvenient.
6. I added an indicator in ChTimestepperHHT to tell the trend of convergence. This is helpful for me in my personal project, to allow me to fall back and do the dynamics iteration again with a more rigorous solver settings. A better solution is to develop this functionality inside ChTimestepperHHT because I think in this way the overall speed is fastest. But I don't know how to develop this fall-back functionality in ChTimestepperHHT, so maybe Prof. Serban could do it?

Please review the PR above. Thanks for your time. Any suggestion would be much appreciated.

### CODE REVIEW

**Comment by rserban** [comment]
Thanks for the PR.
I plan on revisiting HHT at some point and will include additional convergence monitoring.
Just a couple of comments:
- there is no reason to declare the new ChTimestepperHHT  methods as virtual
- getter methods should be declared as const
- there is no need for a semicolon after the closing bracket of a function definition

File: src/chrono/timestepper/ChTimestepperHHT.h [hunk 1]
```diff
@@ -122,6 +125,22 @@ class ChApi ChTimestepperHHT : public ChTimestepperIIorder, public ChImplicitIte
     virtual void Advance(const double dt  ///< timestep to advance
                          ) override;
 
+    /// Get an indicator to tell whether the iteration in current step tends to convergence or divergence.
+    /// This could be helpful if you want to fall back to iterate again via using more rigorous stepper settings, 
+    /// such as smaller fixed time stepper, turning off ModifiedNerton,etc, 
+    /// WHEN you are not satisfied by current iteration result.
+    bool GetConvergenceFlag() { return convergence_trend_flag; };
```
Comment (rserban) [inline src/chrono/timestepper/ChTimestepperHHT.h:132]: Please change to:
bool GetConvergenceFlag() const { return convergence_trend_flag; }

File: src/chrono/timestepper/ChTimestepperHHT.h [hunk 2]
```diff
@@ -122,6 +125,22 @@ class ChApi ChTimestepperHHT : public ChTimestepperIIorder, public ChImplicitIte
...
+
+    /// Set the threshold of norm of R, which is used to judge the trend of convergency.
+    /// For different systems, this threshold may be much different, such as a mini robot and a huge wind turbine.
+    /// You could turn on 'verbose' of HHT stepper and look at the norm of R for your system firstly, 
+    /// and then set a suitable threshold of norm of R. 
+    /// This threshold should always be larger than the normal norm of R for your system.
+    void SetThreshold_R(double mv) { threshold_R = mv; };
```
Comment (rserban) [inline src/chrono/timestepper/ChTimestepperHHT.h:139]: Please change to:
void SetThreshold_R(double mv) { threshold_R = mv; }

File: src/chrono/timestepper/ChTimestepperHHT.h [hunk 3]
```diff
@@ -122,6 +125,22 @@ class ChApi ChTimestepperHHT : public ChTimestepperIIorder, public ChImplicitIte
...
+
+    /// Get the threshold of norm of R, which is used to judge the trend of convergency.
+    double GetThreshold_R() { return threshold_R; };
```
Comment (rserban) [inline src/chrono/timestepper/ChTimestepperHHT.h:142]: Please change to:
double GetThreshold_R() const { return threshold_R; }

Comment (pchaoWT) [reply src/chrono/timestepper/ChTimestepperHHT.h]: See commit 03d74a2 please.

File: src/chrono/timestepper/ChTimestepperHHT.h (hunk 1 above)
Comment (pchaoWT) [reply src/chrono/timestepper/ChTimestepperHHT.h]: See commit 03d74a2 please.

File: src/chrono/timestepper/ChTimestepperHHT.h (hunk 2 above)
Comment (pchaoWT) [reply src/chrono/timestepper/ChTimestepperHHT.h:139]: See commit 9e37f48 please.

**Pull request merged**